File: leetcode_problems/typescript/src/lc1544_make_string_great.ts
function makeGood(s: string): string {
  let res: string[] = [];
  for (let char of s) {
    if (res.length === 0) {
      res.push(char);
      continue;
    }

    let prev = res[res.length - 1];
    if (prev !== char && prev.toLowerCase() === char.toLowerCase()) {
      res.pop();
    } else {
      res.push(char);
    }
  }
  return res.join("");
}

File: leetcode_problems/cpp/lc1544_make_string_great.cpp
#include <string>
#include <algorithm>
using std::string;

class Solution
{
public:
    string makeGood(string s)
    {
        string res;
        for (char chr : s)
        {
            if (res.size() == 0)
            {
                res.push_back(chr);
                continue;
            }

            char prev = res[res.size() - 1];
            if (prev != chr && std::tolower(prev) == std::tolower(chr))
            {
                res.pop_back();
            }
            else
            {
                res.push_back(chr);
            }
        }
        return res;
    }
};